packages/ui/src/hooks/utils.ts

```diff
@@ -126,3 +126,63 @@ export const filterSocialConnectors = (socialConnectors?: ConnectorData[]) => {
 
   return Array.from(connectorMap.values());
 };
+
+/**
+ * Social Connectors Filter Utility Methods used in preview mode only
+ */
+export const filterPreviewSocialConnectors = (
+  previewPlatform: Platform,
+  socialConnectors?: ConnectorData[]
+) => {
+  if (!socialConnectors) {
+    return [];
+  }
+
+  const connectorMap = new Map<string, ConnectorData>();
+
+  /**
+   * Browser Environment
+   * Accepts both web and universal platform connectors.
+   * Insert universal connectors only if there is no  web platform connector provided with the same target.
+   * Web platform has higher priority.
+   **/
+
+  if (previewPlatform === 'web') {
+    for (const connector of socialConnectors) {
+      const { platform, target } = connector;
+
+      if (platform === 'Native') {
+        continue;
+      }
+
+      if (platform === 'Web' || !connectorMap.get(target)) {
```

Comment:
This is a loose check. Currently if the platform is 'web', it would be set into the map even though it is already in the map, which is unnecessary

Should be updated to either
```typescript
if (!connectorMap.get(target))
```

Or to be more strict:

```typescript
if ((platform === 'web' || platform === '') && !connectorMap.get(target))
```

Reply:
Ah, I see